saya/core/src/storage/mod.rs

```diff
@@ -22,3 +25,120 @@ pub struct BlockWithDa {
     pub height: u64,
     pub da_pointer: DataAvailabilityPointer,
 }
+
+pub enum Step {
+    Snos,
+    Bridge,
+}
+pub enum Query {
+    SnosProof,
+    BridgeProof,
+    BridgeTrace,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum BlockStatus {
+    Mined,
+    SnosPieGenerated,
+    SnosProofSubmitted,
+    SnosProofGenerated,
+    BridgePieSubmitted,
+    BridgePieGenerated,
+    BridgeProofSubmitted,
+    BridgeProofGenerated,
+    VerifiedProof,
+    Settled,
+    Failed,
+}
+impl std::fmt::Display for BlockStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            BlockStatus::Mined => write!(f, "mined"),
+            BlockStatus::SnosPieGenerated => write!(f, "snos_pie_generated"),
+            BlockStatus::SnosProofSubmitted => write!(f, "snos_proof_submitted"),
+            BlockStatus::SnosProofGenerated => write!(f, "snos_proof_generated"),
+            BlockStatus::BridgePieSubmitted => write!(f, "bridge_pie_submitted"),
+            BlockStatus::BridgePieGenerated => write!(f, "bridge_pie_generated"),
+            BlockStatus::BridgeProofSubmitted => write!(f, "bridge_proof_submitted"),
+            BlockStatus::BridgeProofGenerated => write!(f, "bridge_proof_generated"),
+            BlockStatus::VerifiedProof => write!(f, "verified_proof"),
+            BlockStatus::Settled => write!(f, "settled"),
+            BlockStatus::Failed => write!(f, "failed"),
+        }
+    }
+}
+
+impl From<&str> for BlockStatus {
+    fn from(s: &str) -> Self {
+        match s {
+            "mined" => BlockStatus::Mined,
+            "snos_pie_generated" => BlockStatus::SnosPieGenerated,
+            "snos_proof_submitted" => BlockStatus::SnosProofSubmitted,
+            "snos_proof_generated" => BlockStatus::SnosProofGenerated,
+            "bridge_pie_submitted" => BlockStatus::BridgePieSubmitted,
+            "bridge_pie_generated" => BlockStatus::BridgePieGenerated,
+            "bridge_proof_submitted" => BlockStatus::BridgeProofSubmitted,
+            "bridge_proof_generated" => BlockStatus::BridgeProofGenerated,
+            "verified_proof" => BlockStatus::VerifiedProof,
+            "failed" => BlockStatus::Failed,
+            "settled" => BlockStatus::Settled,
+            _ => panic!("Invalid block status"),
+        }
+    }
+}
+
+pub trait PersistantStorage {
```

Comment:
Some blank lines between each function could help reading/navigating through the code.